src/GutenbergCleanup/ShortcodeHooks.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace WPML\PB\GutenbergCleanup;
+
+use WPML\FP\Fns;
+use WPML\FP\Maybe;
+use WPML\FP\Obj;
+use function WPML\FP\pipe;
+use function WPML\FP\tap as tap;
+
+class ShortcodeHooks implements \IWPML_Backend_Action {
+
+	public function add_hooks() {
+		add_action(
+			'wp_insert_post',
+			Fns::withoutRecursion( Fns::noop(), [ $this, 'removeGutenbergFootprint' ] ),
+			10, 2
+		);
+	}
+
+	/**
+	 * @param int      $post_ID
+	 * @param \WP_Post $post
+	 */
+	public function removeGutenbergFootprint( $post_ID, \WP_Post $post ) {
+		// $isBuiltWithShortcodes :: \WP_Post -> bool
+		$isBuiltWithShortcodes = function( \WP_Post $post ) {
```

Comment:
@strategio Maybe:

```php
$isBuiltWithShortcodes = partial( 'apply_filters', 'wpml_pb_is_post_built_with_shortcodes', false );
```

Reply:
I also had this in mind but I decided to opt for the verbose form to make it more readable (and findable) for non-fp-literate people.